src/codecs/__mocks__/ssf.ts

```diff
@@ -4,10 +4,20 @@
 
 import { Node } from '@stencila/schema'
 import { create, VFile } from '../../util/vfile'
+import { Codec } from '../types'
 
 export const fileNames = ['super-special-file']
 export const extNames = ['ssf']
 export const mediaTypes = ['application/vnd.super-corp.super-special-file']
 export const sniff = async (content: string) => /^SSF:/.test(content)
 export const decode = async (file: VFile) => null
 export const encode = async (node: Node, options: {} = {}) => create()
+
+export class Ssf extends Codec implements Codec {
```

Comment:
Why do all the codecs both `extend` and `implement` `Codec`? Shouldn't just extending accomplish the method checking?

Reply:
It has to do with how the implementation of properties are enforced and the ability to check class inheritance at runtime.

<img width="1583" alt="Screen Shot 2019-08-12 at 13 19 03" src="https://user-images.githubusercontent.com/1646307/62884193-e7061d80-bd03-11e9-8659-1cabae66bea7.png">

Reply:
Thanks for detailed explanation @alex-ketch - if you still have those demo snippets available it might be worth putting them in a `docs/developing-a-codec.md`:

```markdown
## FAQ

....

#### Why do all the codecs both extend and implement Codec? Shouldn't just extending accomplish the method checking?

It has to do with how the implementation of properties are enforced and the ability to check class inheritance at runtime.

....

```

Reply:
Good idea @nokome
Documented in https://github.com/stencila/encoda/pull/217/commits/88d6152ed498239b9f0ef03cc1cab311056d89ea